SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestTearOff/SpatialTestTearOff.cpp

```diff
@@ -0,0 +1,186 @@
+ *  - The 2 spawned Actors are deleted.
+ */
+
+ASpatialTestTearOff::ASpatialTestTearOff()
+	: Super()
+{
+	Author = "Andrei";
+	Description = TEXT("Test TearOff prevents Actors from replicating");
+
+	SpwanedReplicatedActorBaseLocation = FVector(150.0f, 150.0f, 80.0f);
+}
+
+void ASpatialTestTearOff::BeginPlay()
+{
+	Super::BeginPlay();
+
+	// Step definition for a 5 second timer
+	FSpatialFunctionalTestStepDefinition WorkerWaitForTimeStepDefinition;
+	WorkerWaitForTimeStepDefinition.bIsNativeDefinition = true;
+	WorkerWaitForTimeStepDefinition.TimeLimit = 15.0f;
+	WorkerWaitForTimeStepDefinition.NativeStartEvent.BindLambda([this]()
+	{
+		TimerHelper = 0.0f;
+	});
+	WorkerWaitForTimeStepDefinition.NativeTickEvent.BindLambda([this](float DeltaTime)
+	{
+		TimerHelper += DeltaTime;
+		if (TimerHelper > 5.0f)
+		{
+			FinishStep();
+		}
+	});
+
+	// All workers set a reference to the ReplicatedTearOffActor placed in the map.
+	AddStep(TEXT("SpatialTestTearOffUniversalStartupActorReferenceSetup"), FWorkerDefinition::AllWorkers, nullptr, [this]()
+		{
+			TArray<AActor*> ReplicatedTearOffActors;
+			UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedTearOffActor::StaticClass(), ReplicatedTearOffActors);
+
+			checkf(ReplicatedTearOffActors.Num() == 1, TEXT("Exactly one ReplicatedTearOffActor is expected at this stage"));
+
+			StartupTearOffActor = Cast<AReplicatedTearOffActor>(ReplicatedTearOffActors[0]);
+			if (IsValid(StartupTearOffActor))
+			{
+				FinishStep();
+			}
+		});
+
+	// This is required to make the test pass in Native since calling TearOff does not immediately stop the Actor from replicating.
+	AddStepFromDefinition(WorkerWaitForTimeStepDefinition, FWorkerDefinition::AllWorkers);
+
+	// The server changes the value of the replicated property of the ReplicatedTearOffActor that was initially placed in the map.
+	AddStep(TEXT("SpatialTestTearOffServerChangeStartupActorDefaultProperty"), FWorkerDefinition::Server(1), nullptr, [this]()
+		{
+			if (StartupTearOffActor)
+			{
+				StartupTearOffActor->TestInteger = 1;
+				FinishStep();
+			}
+		});
+
+	// Allow for potential replication to propagate before checking that the property update was ignored.
+	AddStepFromDefinition(WorkerWaitForTimeStepDefinition, FWorkerDefinition::AllWorkers);
+
+	// Clients check that the server update to the replicated property was ignored, as expected.
+	AddStep(TEXT("SpatialTestTearOffClientsCheckStartupActorPropertyValue"), FWorkerDefinition::AllClients, nullptr, [this]()
+		{
+			if (StartupTearOffActor && StartupTearOffActor->TestInteger == 0)
+			{
+				FinishStep();
+			}
+		});
+
+	// The server dynamically spawns a ReplicatedTearOffActor.
+	AddStep(TEXT("SpatialTestTearOffServerSpawnReplicatedTearOffActor"), FWorkerDefinition::Server(1), nullptr, [this]()
+		{
+			AReplicatedTearOffActor* SpawnedTearOffActor = GetWorld()->SpawnActor<AReplicatedTearOffActor>(FVector(50.0f, 50.0f, 80.0f), FRotator::ZeroRotator, FActorSpawnParameters());
```

Comment:
I see this is just coded here. Since there is a spawn & movement flow, I think I would slighly prefer all of these locations to be in variables and declared somewhere close together.

Reply:
Sure thing, will add this!